### ISSUE TITLE
2.1.1/1.3.9 "sound feedback" generate no sound in pass-through mode

### ISSUE THREAD

**Issue by father-mande** [plexinc/plex-media-player#621]
Hi,

Users on QNAP NAS (CPU Intel J1900 IGP HD used) have problem with pass-through to A.V.R. or sound bar
PMP 2.1.1 or PMP 1.3.9 (same platform the two version can coexist on NAS (QNAP or Asustor)
ex. using YSP-5600
sound feedback checked / pass-through DTS/DTS HD Master/DTS:X : NO SOUND
uncheck "sound feedback" restart PMP (mandatory don't know why ... but)
same movies pass-through DTS/DTS HD Master/DTS:X : sound works as expected

Q? what is the relation between "sound feedback" (navigation sound) and pass-through ??? is it a bug ?

In same context  trueHD / Atmos don't produce sound ... even YSP-5600 switch to True HD, but perhaps it's another problem (profiles) ... BUT due to the fact tests will be done in same context ... I also inform you of this  ... if you have some advice ...

Thanks.
Philippe.
NB I open this issue, user hesitate to do it itself, but I think somebody with an AVR can confirm or infirm the problem.
NB1 PMP 1.3.9 & PMP 2.1.1 have been build from "releases" github + QT5.9 on ubuntu 14.04 MPV last github update and conan updated (all fresh build)

**Comment by wm4** [comment]
Sound should work fine outside of playback regardless of pass-through settings. If it doesn't work, there's a possibility that Qt selects the wrong sound card or so.

Sound feedback during playback won't work with passthrough of course. It's impossible without either decoding the audio and encoding it again (defeats the point of passthrough), or without a second audio device and output hardware. So if that is the reason, it won't be fixed. Note that playing music with PMP still counts as playback, even if it's in the background.

**Comment by father-mande** [comment]
Hi,

Thanks ...
So, if I understand ... user MUST suppress "sound feedback" to avoid faulty choice for output sound device ... and this can't be automated when any pass-trough is selected in settings ....

Because as you said (confirming users tests), sound work great but without "sound feedback" ...

Do you think that returning to Qt5.7 (actual is Qt5.9 can solve this problem ???
Apologize for supposition, I don't have an AVR to test myself .

Thanks again.
Philippe.

**Comment by Paulo-Cerdeira** [comment]
Hi!
I'm the user that @father-mande  mentioned on firs post!

@wm4 am I wrong or you understand the opposite of what we are talking??!
It seems you thought that we wan't the navigation sounds during playback (on pause screen) when the output is set to passthrough! Or I understand everything wrong??
What we want is the output sound working at passthrough mode independently of sound feedback is set or not (that's what father-mande requested to be fixed, eventually with the workaround of auto disable the sound feedback when passthrough is set)...

Many thanks,
Cerdeira

**Comment by wm4** [comment]
>It seems you thought that we wan't the navigation sounds during playback (on pause screen) when the output is set to passthrough!

Yes.

> What we want is the output sound working at passthrough mode independently of sound feedback is set or not (that's what father-mande requested to be fixed, eventually with the workaround of auto disable the sound feedback when passthrough is set)...

Oh. these shouldn't conflict. What could possibly happen is that the code for navigation sounds (i.e. Qt) keeps the audio connection open for a while, and if it's still open when playback starts, audio initialization for playback would fail and fall back to non-exclusive PCM.

Cna you post two PMP logs please, one where passthrough works, and one where it doesn't? (1 playback attempt each should be good.)

**Comment by Paulo-Cerdeira** [comment]
Hi again!

here are the logs:

[plexmediaplayer-work.log](https://github.com/plexinc/plex-media-player/files/1375408/plexmediaplayer-work.log)
[plexmediaplayer-nosound.log](https://github.com/plexinc/plex-media-player/files/1375409/plexmediaplayer-nosound.log)

I analyse them and I saw these when sound playback fails:

ao/alsa: opening device 'hdmi:CARD=PCH,DEV=1' => 'hdmi:CARD=PCH,DEV=1,AES0=6,AES1=130,AES2=0,AES3=2' 
2017-10-11 12:18:15
[ ERROR ] PlayerComponent.cpp @ 532 - ao/alsa: Playback open error: Device or resource busy 
2017-10-11 12:18:15 [ ERROR ] PlayerComponent.cpp @ 532 - ao: Failed to initialize audio driver 'alsa' 
2017-10-11 12:18:15 [ ERROR ] PlayerComponent.cpp @ 532 - ao: This audio driver/device was forced with the --audio-device option.

**Comment by wm4** [comment]
Yep, that sounds like that is happening. Can you test whether enabling sound feedback, navigating a bit around (so they're played), disabling them, and then playing something works?

**Comment by Paulo-Cerdeira** [comment]
No, it doesn’t work. After disabling it (the sound feedback), I’ve to restart PMP to get sound working. I already test enabling it, navigate through the play button, wait a few, and then only press it - still no sound

**Comment by LongChair** [comment]
@father-mande : is this one still relevant ?

**Comment by father-mande** [comment]
Hi,

If the user Paulo-cerdeira don't follow this ...  and now PMP 1 is stooped and PMP 2 is at version 2.8.0, I suppose you can close this post ... Paulo will reopen it if the error is always present ...

Philippe.
